cdc_vaccines/delphi_cdc_vaccines/pull.py

```diff
@@ -0,0 +1,173 @@
+
+    drop_columns =  list(set(drop_columns + [x for x in df.columns if
+        ("pct" in x) | ("svi" in x)] + list(df.columns[22:])))
+    df = GeoMapper().add_geocode(df, "state_id", "state_code",
+        from_col="recip_state", new_col="state_id", dropna=False)
+    df['state_id'] = df['state_id'].fillna('0').astype(int)
+    # Change FIPS from 0 to XX000 for statewise unallocated cases/deaths
+    unassigned_index = (df["fips"] == "UNK")
+    df.loc[unassigned_index, "fips"] = df["state_id"].loc[unassigned_index].values * 1000
+
+    # Conform FIPS
+    df["fips"] = df["fips"].apply(lambda x: f"{int(x):05d}")
+    df["timestamp"] = pd.to_datetime(df["date"])
+    # Drop unnecessary columns (state is pre-encoded in fips)
+    try:
+        df.drop(drop_columns, axis=1, inplace=True)
+    except KeyError as e:
+        raise ValueError(
+            "Tried to drop non-existent columns. The dataset "
+            "schema may have changed.  Please investigate and "
+            "amend drop_columns."
+        ) from e
+
+    # timestamp: str -> datetime
+    try:
+        df.columns = ["fips",
+                      "cumulative_counts_tot_vaccine",
+                      "cumulative_counts_tot_vaccine_12P",
+                      "cumulative_counts_tot_vaccine_18P",
+                      "cumulative_counts_tot_vaccine_65P",
+                      "cumulative_counts_part_vaccine",
+                      "cumulative_counts_part_vaccine_12P",
+                      "cumulative_counts_part_vaccine_18P",
+                      "cumulative_counts_part_vaccine_65P",
+                      "timestamp"]
+    except KeyError as e:
+        raise ValueError(
+            "Tried to name wrong number of columns. The dataset "
+            "schema may have changed.  Please investigate and "
+            "amend drop_columns."
+        ) from e
+
+    df_dummy = df.loc[(df["timestamp"] == min(df["timestamp"]))].copy()
+    df_dummy.loc[:, "timestamp"] = df_dummy.loc[:, "timestamp"] - pd.Timedelta(days=1)
+    df_dummy.loc[:, ["cumulative_counts_tot_vaccine",
+                    "cumulative_counts_tot_vaccine_12P",
+                    "cumulative_counts_tot_vaccine_18P",
+                    "cumulative_counts_tot_vaccine_65P",
+                    "cumulative_counts_part_vaccine",
+                    "cumulative_counts_part_vaccine_12P",
+                    "cumulative_counts_part_vaccine_18P",
+                    "cumulative_counts_part_vaccine_65P",
+                    ]] = 0
+
+    df =pd.concat([df_dummy, df])
+    # Obtain new_counts
+    df.sort_values(["fips", "timestamp"], inplace=True)
+    for to, from_d in DIFFERENCE_MAPPING.items():
+        df[to] = df[from_d].diff()
+
+    rem_list = [ x for x in list(df.columns) if x not in ['timestamp', 'fips'] ]
+    # Handle edge cases where we diffed across fips
+    mask = df["fips"] != df["fips"].shift(1)
+    df.loc[mask, rem_list] = np.nan
+    df.reset_index(inplace=True, drop=True)
```

Comment:
I recommend using this instead, since it's clearer and uses more pandas built-ins (it's already used in [JHU](https://github.com/cmu-delphi/covidcast-indicators/blob/4ac71f7fd1feb15058f483b2cbd374f7ed208128/jhu/delphi_jhu/pull.py#L37))

```suggestion
    df = df.set_index(["fips", "timestamp"])
    for to, from_d in DIFFERENCE_MAPPING.items():
        df[to] = df.groupby(level=0)[from_d].diff()
   df.reset_index(inplace=True)
```

I compared this with your existing code and these all passed fine (df is from existing code, df2 is from the suggestion)
```
    df = df.set_index(["fips", "timestamp"])
    for column in df.columns:
        if "incidence" in column:
            pd.testing.assert_series_equal(df[column], df2[column])
```